### ISSUE TITLE
`kubectl top --sort-by` is not sorting correctly

### ISSUE THREAD

**Issue by k-vishwa** [kubernetes/kubernetes#81270]
The --sort-by option to kubectl top was introduced from v1.15.2 and it accepts only 2 params i.e. cpu and memory.

e.g. 
kubectl top po --all-namespaces --sort-by=cpu
kubectl top po --all-namespaces --sort-by=memory

The output is not in a sorted manner.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.15.2
- Cloud provider: GCP/GKE

**Comment by k-vishwa** [comment]
@kubernetes/sig-cli

**Comment by tedyu** [comment]
Do you have sample output with incorrect order ?

**Comment by zouyee** [comment]
/sig cli
@kubernetes/sig-cli-bugs

**Comment by k-vishwa** [comment]
Here is the output 
`
bash ~ $ kubectl version --short
Client Version: v1.15.2
Server Version: v1.11.10-gke.5
bash ~ $

bash ~ $ k top po -n kube-system --sort-by=cpu
NAME                                                             CPU(cores)   MEMORY(bytes)
kube-proxy-gke-walletservices-prd-u-default-pool-1319e1cf-44wh   7m           21Mi
fluentd-gcp-v3.2.0-vsnkz                                         181m         420Mi
kube-dns-76dbb796c5-bc2jq                                        2m           40Mi
fluentd-gcp-v3.2.0-fwmds                                         199m         307Mi
kube-dns-76dbb796c5-tfqj2                                        2m           40Mi
kube-proxy-gke-walletservices-prd-u-default-pool-fec415af-rt90   7m           20Mi
fluentd-gcp-v3.2.0-m92j4                                         260m         427Mi
fluentd-gcp-v3.2.0-8l6mt                                         12m          213Mi
metrics-server-v0.2.1-fd596d746-h4dfr                            2m           55Mi
kube-proxy-gke-walletservices-prd-u-default-pool-8d1d2435-5jq0   8m           21Mi
heapster-v1.6.0-beta.1-66f9c4d4d9-9b54q                          1m           67Mi
fluentd-gcp-scaler-5b5bf8b99f-lgdxk                              21m          112Mi
kube-proxy-gke-walletservices-prd-u-default-pool-fec415af-gv7s   6m           21Mi
prometheus-to-sd-ptqbg                                           0m           15Mi
prometheus-to-sd-26r6w                                           0m           15Mi
prometheus-to-sd-7z7cs                                           0m           15Mi
prometheus-to-sd-7n9rs                                           0m           15Mi
event-exporter-v0.2.4-5c57bb7478-p7smw                           0m           27Mi
l7-default-backend-7ff48cffd7-mn6gw                              0m           3Mi
kube-dns-autoscaler-67c97c87fb-wrdc9                             0m           10Mi
bash ~ $`

**Comment by k-vishwa** [comment]
Also MEMORY sorting is not correct. Please fix that too along with CPU sort.

**Comment by k-vishwa** [comment]
Still the fix is not made in 1.16 kubectl version

```
bash ~ $ ./kubectl top po -n kube-system --sort-by=cpu                                                                             [ruby-2.3.7p456]
NAME                               CPU(cores)   MEMORY(bytes)
kube-scheduler-minikube            3m           11Mi
coredns-5644d7b6d9-sxrfr           2m           7Mi
etcd-minikube                      59m          25Mi
coredns-5644d7b6d9-jjzdz           4m           5Mi
kube-addon-manager-minikube        184m         81Mi
kube-controller-manager-minikube   99m          13Mi
kube-apiserver-minikube            97m          247Mi
kube-proxy-vggt5                   0m           10Mi
storage-provisioner                0m           14Mi
metrics-server-587f876775-wkn4r    1m           10Mi
bash ~ $ ./kubectl version --short                                                                                                 [ruby-2.3.7p456]
Client Version: v1.16.1
Server Version: v1.16.0
bash ~ $

**Comment by AmazingTurtle** [comment]
I can confirm

```
$ kubectl version --short
Client Version: v1.15.0
Server Version: v1.15.7-gke.2

$ kubectl top pods --sort-by=cpu --all-namespaces                                                                       NAMESPACE     NAME                                                          CPU(cores)   MEMORY(bytes)
kube-system   fluentd-gcp-v3.1.1-nkf2m                                      9m           142Mi
kube-system   kube-proxy-gke-heimdall-cluster-heimdall-pool-8ebbd61e-b0q4   1m           13Mi
kube-system   stackdriver-metadata-agent-cluster-level-8644c9b5c7-7hhrc     3m           20Mi
kube-system   fluentd-gcp-v3.1.1-wh7xr                                      10m          148Mi
kube-system   kube-dns-5dbbd9cc58-dnl84                                     3m           29Mi
kube-system   prometheus-to-sd-vjr6n                                        3m           13Mi
kube-system   fluentd-gcp-v3.1.1-2z62d                                      6m           137Mi
kube-system   kube-proxy-gke-heimdall-cluster-heimdall-pool-8ebbd61e-7hd6   2m           13Mi
kube-system   kube-dns-autoscaler-6b7f784798-tp7p2                          1m           4Mi
kube-system   prometheus-to-sd-kxdnw                                        4m           13Mi
kube-system   l7-default-backend-84c9fcfbb-zjc78                            1m           2Mi
kube-system   prometheus-to-sd-9ff5n                                        3m           13Mi
kube-system   metrics-server-v0.3.3-6d96fcc55-45r8q                         1m           20Mi
kube-system   fluentd-gcp-v3.1.1-tngnw                                      9m           137Mi
kube-system   kube-proxy-gke-heimdall-cluster-heimdall-pool-8ebbd61e-22sz   1m           14Mi
kube-system   heapster-595cf6c666-5v99x                                     2m           38Mi
kube-system   kube-proxy-gke-heimdall-cluster-heimdall-pool-8ebbd61e-ndlk   2m           14Mi
kube-system   kube-dns-5dbbd9cc58-mtfmw                                     3m           29Mi
kube-system   event-exporter-v0.3.0-74bf544f8b-g4lvq                        1m           20Mi
kube-system   fluentd-gcp-scaler-dd489f778-qvh2x                            0m           35Mi
kube-system   prometheus-to-sd-94wkf                                        2m           12Mi
```

**Comment by Aracki** [comment]
Just a little hack until this is fixed:
`kubectl top pod --all-namespaces | sort --reverse --key 4 --numeric`

**Comment by sbl-matter** [comment]
So, the sorting options is broken only when option `--all-namespaces / -A` is used.

**Comment by fergusontom** [comment]
Why was this closed?

**Comment by TBBle** [comment]
To be clear, this was fixed in 1.18 (#82333).
